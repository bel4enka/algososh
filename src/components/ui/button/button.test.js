import renderer from 'react-test-renderer';
import {Button} from "./button";
import {render, screen, fireEvent} from '@testing-library/react'

describe('Работа компонента - Button', () => {
    it('Кнопка с текстом', () => {
        const button = renderer
            .create(<Button text='Текст'/>)
            .toJSON()
        expect(button).toMatchSnapshot()
    })
    it('Кнопка без текста', () => {
        const button = renderer
            .create(<Button />)
            .toJSON()
        expect(button).toMatchSnapshot()
    })
    it('Заблокированная кнопка', () => {
        const button = renderer
            .create(<Button disabled={true}/>)
            .toJSON()
        expect(button).toMatchSnapshot()
    })
    it('Кнопка с индикацией загрузки', () => {
        const button = renderer
            .create(<Button isLoader={true}/>)
            .toJSON()
        expect(button).toMatchSnapshot()
    })
    it('Колбек кнопки', () => {
        window.alert = jest.fn();

        // Рендерим компонент
        render(<Button text="Рецепт пельменей" onClick={()=>{alert('Ура! Пельмени!')}}/>)

        // Находим элемент ссылки
        const link = screen.getByText("Рецепт пельменей");

        // Имитируем нажатие на ссылку
        fireEvent.click(link);

        // Проверяем, что alert сработал с правильным текстом предупреждения
        expect(window.alert).toHaveBeenCalledWith('Ура! Пельмени!');
    });
})