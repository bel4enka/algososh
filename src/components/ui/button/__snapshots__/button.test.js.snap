// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Работа компонента - Button Заблокированная кнопка 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Работа компонента - Button Заблокированная кнопка 2`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Работа компонента - Button Кнопка без текста 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Работа компонента - Button Кнопка без текста 2`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Работа компонента - Button Кнопка с индикацией загрузки 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Работа компонента - Button Кнопка с текстом 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Текст
  </p>
</button>
`;
